lib/web/databases.go

```diff
@@ -644,7 +646,8 @@ func (s *databaseInteractiveSession) Run() error {
 	// handler.
 	go func() {
 		alpnConnWithAddr := utils.NewConnWithAddr(s.alpnConn, s.ws.LocalAddr(), s.ws.RemoteAddr())
-		if err := s.alpnHandler(s.ctx, alpnConnWithAddr); !utils.IsOKNetworkError(err) {
+		ctx := alpncommon.WithConnHandlerSource(s.ctx, alpncommon.ConnHandlerSourceWebDB)
+		if err := s.alpnHandler(ctx, alpnConnWithAddr); !utils.IsOKNetworkError(err) {
```

Comment:
nit: 

Passing the connection type via ctx to the handler function feels like an overuse of context. Both the connection param type and the handler method are inherently tied to the connection itself.

Would it be feasible to pass the connection type as an explicit argument (or optional param), or to extend the connection interface to include a conn.Type() method?

If that adds too much complexity to the scope, I’m fine with the current approach of passing it via context -  I just want to confirm that making it explicit was considered.